### PR TITLE
fix wrong skip_hint on hashindex_set when encountering tombstones

### PR DESCRIPTION
**Pull request by rciorba**
hashindex_lookup would always hint at skipping whatever it's probe
length had been with no regard for tombstones it had encountered. This
meant new keys would not overwrite first tombstones, but would always
land on empty buckets. 

Ran the benchmarks I used for the Robin Hood hashing so you can 
compare the impact of this patch:
https://cdn.rawgit.com/rciorba/d8990a54be8b778bb3322ac40364de90/raw/90d05773cc377b061f400d3833166bbce4243ba1/results.html

That is a regression I introduced myself as part of: #1748

### CODE REVIEW

File: src/borg/_hashindex.c [hunk 1]
```diff
@@ -383,18 +383,18 @@ hashindex_get(HashIndex *index, const void *key)
 static int
 hashindex_set(HashIndex *index, const void *key, const void *value)
 {
-    int offset = 0;
-    int idx = hashindex_lookup(index, key, &offset);
+    int start_addr;
```
Comment (ThomasWaldmann) [inline src/borg/_hashindex.c]: use start_hint? or start_idx?
*_addr sounds rather like a memory address / pointer (see BUCKET_ADDR()).

Comment (rciorba) [inline src/borg/_hashindex.c]: Good point! `start_idx` it is.

**Comment by ThomasWaldmann** [comment]
@enkore can you have a look also?

**Comment by ThomasWaldmann** [comment]
@rciorba thanks!

**Pull request merged**